components/filter-components/Timeline.vue

```diff
@@ -41,29 +43,61 @@ export default {
   },
   data() {
     return {
-      rangeValue: [this.startRange[0].toString(), this.startRange[1].toString()],
       dates: this.timelineDates,
       markers: this.timelineMarkers,
-      timelineShowing: true
+      timelineShowing: true,
+      numberRangeValue: [0, 0] // Initialize with dummy values
     }
   },
+  mounted() {
+    // Set initial range after `max` is computed
+    this.numberRangeValue = [0, this.max];
+  },
   watch: {
-    rangeValue(newRange, oldQuestion) {
-      this.$emit("updatedRangeValue", newRange);
+    numberRangeValue(newRange) {
+      // Convert the slider range values to Date objects
+      const dateRange = [
+        this.parseDate(this.dates[newRange[0]].label),
+        this.parseDate(this.dates[newRange[1]].label)
+      ];
+      this.$emit("updatedRangeValue", dateRange);
+      console.log("Updated Date Range", dateRange);
+    }
+  },
+  computed: {
+    max() {
+      return this.dates.length - 1;
     }
   },
   methods: {
-    emitRange: function (rangeValue) {
+    emitRange(rangeValue) {
       this.$emit("updatedRangeValue", rangeValue);
     },
     toggleTimelineShowing() {
       this.timelineShowing = !this.timelineShowing;
       this.$emit('timelineToggled', this.timelineShowing);
+    },
+    getDateLabel(value) {
+      return this.dates[value].label;
+    },
+    handleCustomInputRangeChange(newCustomInputRange) {
+      this.numberRangeValue = newCustomInputRange;
+      console.log("New Number range value", newCustomInputRange);
+    },
+    parseDate(dateStr) {
+      // Convert DD/MM/YYYY to Date object
```

Comment:
Think we can remove this comment as it isn't adding much but the one below is fantastic!

Reply:
Confirmed as removed